packages/core/contracts/insured-bridge/ovm/Optimism_Wrapper.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title Optimism Eth Wrapper
+ * @dev Any ETH sent to this contract is wrapped into WETH and sent to the set bridge pool. This enables ETH to be sent
+ * over the canonical Optimism bridge, which does not support WETH bridging.
+ */
+interface WETH9Like {
+    function deposit() external payable;
+
+    function transfer(address guy, uint256 wad) external;
+
+    function balanceOf(address guy) external view returns (uint256);
+}
+
+contract Optimism_Wrapper is Ownable {
+    WETH9Like weth;
+    address bridgePool;
+
+    constructor(WETH9Like _weth, address _bridgePool) {
+        weth = _weth;
+        bridgePool = _bridgePool;
+    }
+
+    function changeBridgePool(address newBridgePool) public onlyOwner {
+        bridgePool = newBridgePool;
+    }
+
+    receive() external payable {
+        wrapAndTransfer();
+    }
+
+    fallback() external payable {
+        wrapAndTransfer();
+    }
```

Comment:
Do we need to define both of these? Any reason not to _only_ define the payable fallback function?

Reply:
True. It'll likely work with just the one. However, if there is data from the transfer then we need the other one.

Reply:
`receive() external payable` — for empty calldata (and any value)
`fallback() external payable` — when no other function matches (not even the receive function). Optionally payable.

Having both means that if the bridge includes data (it can do this) we'll still capture the eth. This means that no matter how funds are transferred into the contract we'll wrap them and send them to the pool.

Reply:
Can you add these as comments

Reply:
done!

Reply:
@chrismaree the docs seem to say that a payable fallback function can handle _all_ cases (incoming ETH or calldata): https://docs.soliditylang.org/en/v0.8.10/contracts.html?highlight=fallback#fallback-function. I don't think we need both. Not a huge issue, though.